peps/pep-0735.rst

```diff
@@ -0,0 +1,650 @@
+path multiple times (for example, via two combined includes) using distinct
+Path Dependencies.
+
+If the ``path`` values are distinct, but refer to the same concrete file or
+directory on the filesystem, handling behavior is unspecified. Implementations
+MAY normalize these paths to a single value.
+
+If the ``path`` values are identical, implementations MUST treat the result as
+a singular Path Dependency following the rules below:
+
+- The ``extras`` keys are concatenated into a single list
+- If ``editable`` is always ``true`` or always ``false``, it is treated as
+  such. Otherwise, the ``editable`` key is treated as though it were absent.
+- If the ``only-deps`` key is present, a ``false`` or absent value takes
+  priority over any ``true`` values. In other words, implementations respect
+  ``only-deps`` if it is ``true`` for all instances of the same ``path``.
+  Otherwise, they treat it as ``false``.
+
+
+Example Dependency Groups Table
+-------------------------------
+
+The following is an example of a partial ``pyproject.toml`` which uses this to
+define four dependency groups: ``test``, ``docs``, ``typing``, and
+``typing-test``:
+
+.. code:: toml
+
+    [dependency-groups]
+    test = ["pytest", "coverage", {path = ".", editable = true}]
+    docs = ["sphinx", "sphinx-rtd-theme"]
+    typing = ["mypy", "types-requests", {path = ".", extras = ["types"], only-deps = true}]
+    typing-test = [{include = "typing"}, {include = "test"}, "useful-types"]
+
+    [project.optional-dependencies]
+    types = ["typing-extensions"]
+
+Note how ``test`` and ``typing`` refer to the current package while ``docs``
+does not. This reflects the ability of Dependency Groups to be used in the same
+manner as extras, adding to dependencies, or completely independently.
+
+``typing-test`` is defined as a union of two existing groups, plus an
+additional package. ``typing`` includes an extra, ``types``, and this is
+included by extension under ``typing-test``.
+Under ``typing-test`` implementations may choose whether or not to use an
+editable installation of the current package or not, but they MUST treat
+``only-deps`` as ``false``.
+
+Package Building
+----------------
+
+Build backends MUST NOT include dependency group data in built distributions.
+
+Use of Dependency Groups
+------------------------
+
+Tools which support Dependency Groups are expected to provide new options and
+interfaces to allow users to install from Dependency Groups.
+
+No syntax is defined for expressing the dependency group of a package, for two
+reasons:
+
+* it would not be valid to refer to the dependency groups of a third-party
+  package from PyPI (because the data is defined to be unpublished)
+
+* there is not guaranteed to be a current package for dependency groups -- part
+  of their purpose is to support non-package projects
+
+For example, a possible pip interface for installing dependency groups
```

Comment:
Should this term always be capitalized?

Reply:
I'm not 100% sure, but I think it should be in the PEP.
If the PEP is accepted, we should get used to it appearing without being title-cased, but within the PEP I think it should be consistently title-ized.

Reply:
I've applied the change to title-case it in all circumstances.